pkg/app/ops/insightcollector/deployments.go

```diff
@@ -0,0 +1,455 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package insightcollector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"runtime"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/pipe-cd/pipe/pkg/datastore"
+	"github.com/pipe-cd/pipe/pkg/filestore"
+	"github.com/pipe-cd/pipe/pkg/insight"
+	"github.com/pipe-cd/pipe/pkg/model"
+)
+
+var (
+	pageSize = 50
+)
+
+func (i *InsightCollector) ProcessNewlyCreatedDeployments(ctx context.Context) error {
+	now := time.Now()
+	targetDate := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC)
+	m, err := i.insightstore.LoadMilestone(ctx)
+	if err != nil {
+		if err == filestore.ErrNotFound {
+			m = &insight.Milestone{}
+		} else {
+			return err
+		}
```

Comment:
nit: One thing. I feel like fast-returen would be better.
```suggestion
	if err != nil {
		if !errors.Is(err, filestore.ErrNotFound) {
			return err
		}
		m = &insight.Milestone{}
	}
```